src/__mocks__/axios.js

```diff
@@ -0,0 +1,9 @@
+const axios = jest.genMockFromModule('axios');
+
+const mockAxios = {
```

Comment:
https://8thlight.com/blog/eric-smith/2011/10/27/thats-not-yours.html
https://github.com/testdouble/contributing-tests/wiki/Don't-mock-what-you-don't-own

Reply:
I refactored to create a APIHandler that can take in a Fetcher and MockFetcher class